src/content/docs/identityserver/upgrades/identityserver4-to-duende-identityserver-v7.mdx

```diff
@@ -0,0 +1,487 @@
+   ```
+
+</Steps>
+
+For your configuration data, the conversation from the old schema to the new will need to be performed by applying a custom SQL script.
+We'll start with creating a migration that targets the `ConfigurationDbContext` database context:
+
+<Steps>
+
+1. Create the migration:
+
+   ```bash title="Terminal"
+   dotnet ef migrations add Config_v4 -c ConfigurationDbContext -o Migrations/ConfigurationDb
+   ```
+
+   You will see a message _"An operation was scaffolded that may result in the loss of data. Please review the migration for accuracy."_ in the output.
+   To avoid data loss, the migration will need to be updated.
+
+2. To ensure no data is lost, make sure to include the [`ConfigurationDb_v4_delta.sql`](https://github.com/DuendeArchive/UpgradeSample-IdentityServer4-v3/blob/main/IdentityServerMigrationSample/ConfigurationDb_v4_delta.sql)
+   script in your project.
+
+   You can add the script as an embedded resource by updating the `.csproj` file:
+
+   ```xml title=".csproj"
+   <ItemGroup>
+       <EmbeddedResource Include="ConfigurationDb_v4_delta.sql" />
+   </ItemGroup>
+   ```
+
+   :::note[Update the SQL script for your database type]
+   The `ConfigurationDb_v4_delta.sql` file assumes you are using SQL Server. If a different database server type is used for your IdentityServer host, you'll need to update the SQL script to use the correct syntax.
+   :::
+
+3. Modify the migration class that was just created and replace it with the following code:
+
+   ```csharp title="Config_v4.cs"
+   using System.IO;
+   using Microsoft.EntityFrameworkCore.Migrations;
+
+   namespace IdentityServerMigrationSample.Migrations.ConfigurationDb
+   {
+       public partial class Config_v4 : Migration
+       {
+           protected override void Up(MigrationBuilder migrationBuilder)
+           {
+               var assembly = typeof(Program).Assembly;
+
+               using (var s = assembly.GetManifestResourceStream("IdentityServerMigrationSample.ConfigurationDb_v4_delta.sql"))
+               {
+                   using (StreamReader sr = new StreamReader(s))
+                   {
+                       var sql = sr.ReadToEnd();
+                       migrationBuilder.Sql(sql);
+                   }
+               }
+           }
+
+           protected override void Down(MigrationBuilder migrationBuilder)
+           {
+           }
+       }
+   }
+   ```
+
+4. Apply the migration to your database:
+
+   ```bash title="Terminal"
+   dotnet ef database update -c ConfigurationDbContext
+   ```
+
+</Steps>
+
+Your database schema should now be updated. You can verify all data is intact, using the [queries outlined in `query_v4.sql`](https://github.com/DuendeArchive/UpgradeSample-IdentityServer4-v3/blob/main/IdentityServerMigrationSample/query_v4.sql).
+
+<details>
+    <summary>View SQL queries to validate migration succeeded</summary>
+
+    ```sql title="query_v4.sql"
+    SELECT * FROM __EFMigrationsHistory
```

Comment:
Anders told me some customers could have millions of clients in their database. Doing a ```select *``` might not be the best option here. Should you limit the calls using `Top`?